### PR TITLE
feat: set configuration for base-result-link to ignore tagging of click

### PR DESCRIPTION
**Pull request by CachedaCodes**
[EMP-864](https://searchbroker.atlassian.net/browse/EMP-864)

## Motivation and context
With the future inclusion of the `displayClick` event we needed a way of ignoring the click event of the base-result-link in the tagging if we decided so. This PR adds a property to the wiring metadata to do that and check it in the module's wiring.

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?

New tests have been done for the new features of `BaseResultLink`.

[EMP-864]: https://searchbroker.atlassian.net/browse/EMP-864?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/x-components/src/components/result/base-result-link.vue [hunk 1]
```diff
@@ -65,16 +66,32 @@
...
+       * have extra events so this record pairs each event to its metadata.
+       */
+      const resultLinkMetadataPerEvent = inject<
+        Partial<
+          Record<
+            PropsWithType<XEventsTypes, Result>,
+            Omit<WireMetadata, 'moduleName' | 'origin' | 'location'>
+          >
+        >
+      >('resultLinkMetadataPerEvent', {});
```
Comment (ajperezbau) [inline packages/x-components/src/components/result/base-result-link.vue:80]: What do you think about adding this injection key to the `injection.consts.ts`?
That way we wouldn't repeat this type when we provide the events.

File: packages/x-components/src/components/result/__tests__/base-result-link.spec.ts [hunk 2]
```diff
@@ -63,9 +66,64 @@ describe('testing BaseResultLink component', () => {
...
+    );
+    resultLinkWrapper.vm.$x.on('UserClickedAResult', true).subscribe(resultClickListener);
+    resultLinkWrapper.vm.$x
+      .on('UserClickedResultAddToCart', true)
+      .subscribe(addToCartClickListener);
+    resultLinkWrapper.trigger('click');
+
+    expect(resultClickListener).toHaveBeenCalledWith({
+      eventPayload: result,
+      metadata: expect.objectContaining(injectedResultLinkMetadataPerEvent.UserClickedAResult)
```
Comment (ajperezbau) [inline packages/x-components/src/components/result/__tests__/base-result-link.spec.ts:119]: I would also test here that the metadata doesn't contain the `UserClickedResultAddToCart` one.

File: packages/x-components/src/components/result/__tests__/base-result-link.spec.ts [hunk 3]
```diff
@@ -63,9 +66,64 @@ describe('testing BaseResultLink component', () => {
...
+        localVue,
+        propsData: { result }
+      }
+    );
+    resultLinkWrapper.vm.$x.on('UserClickedAResult', true).subscribe(resultClickListener);
+    resultLinkWrapper.vm.$x
+      .on('UserClickedResultAddToCart', true)
+      .subscribe(addToCartClickListener);
+    resultLinkWrapper.trigger('click');
+
```
Comment (ajperezbau) [inline packages/x-components/src/components/result/__tests__/base-result-link.spec.ts:116]:
```suggestion
    expect(resultClickListener).toHaveBeenCalledTimes(1);
```

File: packages/x-components/src/components/result/__tests__/base-result-link.spec.ts [hunk 4]
```diff
@@ -63,9 +66,64 @@ describe('testing BaseResultLink component', () => {
...
+    expect(resultClickListener).toHaveBeenCalledWith({
+      eventPayload: result,
+      metadata: expect.objectContaining(injectedResultLinkMetadataPerEvent.UserClickedAResult)
+    });
+    expect(addToCartClickListener).toHaveBeenCalledWith({
```
Comment (ajperezbau) [inline packages/x-components/src/components/result/__tests__/base-result-link.spec.ts:121]:
```suggestion
    expect(addToCartClickListener).toHaveBeenCalledTimes(1);
    expect(addToCartClickListener).toHaveBeenCalledWith({
```

File: packages/x-components/src/components/result/__tests__/base-result-link.spec.ts [hunk 5]
```diff
@@ -63,9 +66,64 @@ describe('testing BaseResultLink component', () => {
...
+      eventPayload: result,
+      metadata: expect.objectContaining(
+        injectedResultLinkMetadataPerEvent.UserClickedResultAddToCart
```
Comment (ajperezbau) [inline packages/x-components/src/components/result/__tests__/base-result-link.spec.ts:124]: I would also test here that the metadata doesn't contain the `UserClickedAResult` one.

File: packages/x-components/src/x-modules/tagging/wiring.ts [hunk 6]
```diff
@@ -161,7 +161,8 @@ export function createTrackWire(property: keyof Tagging): Wire<Taggable> {
       taggingInfo.params.location = location as string;
       return taggingInfo;
     }),
-    ({ eventPayload: { tagging } }) => !!tagging?.[property]
+    ({ eventPayload: { tagging }, metadata: { ignoreInModules } }) =>
+      !!tagging?.[property] && !ignoreInModules?.includes(moduleName)
```
Comment (ajperezbau) [inline packages/x-components/src/x-modules/tagging/wiring.ts:165]: Should we do this in the `wires.factory.ts` to make it compatible with all the modules?
Maybe we can explore it in a future refactor.

**Review by ajperezbau — COMMENTED** [review]
Good job! 👌

File: packages/x-components/src/x-modules/tagging/wiring.ts (hunk 6 above)
Comment (CachedaCodes) [reply packages/x-components/src/x-modules/tagging/wiring.ts:165]: I thought about it but right now is better to do it in a case by case basis for now. We would add an if statement to all the wires and only one will use it. Better keeping it in mind for the future if we have more cases.

File: packages/x-components/src/components/result/base-result-link.vue (hunk 1 above)
Comment (CachedaCodes) [reply packages/x-components/src/components/result/base-result-link.vue:80]: I think those only work for the `XInject` and we're not using them anymore in `2.7` so I don't know if we can use those keys in the native `inject`.

**Pull request merged**